core/domain/exp_domain.py

```diff
@@ -3519,3 +3519,226 @@ def to_metadata_dict(self):
             'title': self.title,
             'objective': self.objective,
         }
+
+
+class StateIdMapping(object):
+    """Domain object for state id mapping model."""
+
+    def __init__(
+            self, exploration_id, exploration_version, state_name_to_ids,
+            latest_state_id_used):
+        """Initialize state id mapping model object.
+
+        Args:
+            exploration_id: str. The exploration id whose state names are
+            mapped.
+            exploration_version: int. The version of the exploration.
+            state_name_to_ids: dict. A dictionary mapping a state name to
+                unique ID.
+            latest_state_id_used: int. Last used state id.
+        """
+        self.exploration_id = exploration_id
+        self.exploration_version = exploration_version
+        self.state_name_to_ids = state_name_to_ids
+        self.latest_state_id_used = latest_state_id_used
+
+    @staticmethod
+    def has_state_changed_significantly(old_state, new_state):
+        """Checks whether there is any significant change in old and new version
+        of the given state.
+
+        A siginificant change has happend when interaction id has been changed
+        in new version of the state.
+
+        Args:
+            old_state: dict. Old version of the state.
+            new_state: dict. New version of the state.
+
+        Returns:
+            bool. A boolean indicating whether change is significant or not.
+        """
+        if old_state.interaction.id != new_state.interaction.id:
+            # State has changed significantly is interaction id in new state is
+            # different from old state.
+            return True
```

Comment:
Do we not return False otherwise? If so, maybe something like:

```
return (old_state.interaction.id != new_state.interaction.id)
```

Reply:
Only this one check is required as of now. We may have more in future (highly unlikely though). Done.